kernel/signal.c

```diff
@@ -777,8 +777,17 @@ long myst_signal_deliver(
 
     uint64_t mask = (uint64_t)1 << (signum - 1);
 
-    /* Only deliver if the signal is not being ignored */
-    if ((handler != (uint64_t)SIG_IGN) || (signum == SIGKILL))
+    /* Deliver signal if
+     * 1. signal is SIGKILL, or
+     * 2. handler is not SIG_DFL and signal is not one of [SIGCHLD, SIGCONT,
+     * SIGSTOP, SIGURG, SIGWINCH], and
+     * 3. handler is not SIG_IGN
+     */
+    if (signum == SIGKILL ||
+        (!(handler == (uint64_t)SIG_DFL &&
+           (signum == SIGCHLD || signum == SIGCONT || signum == SIGSTOP ||
```

Comment:
The `SIGCHLD` can be sent by mystikos kernel. See https://github.com/deislabs/mystikos/blob/063cefb00e514f9899e9ae8022d2e18567b3c9f2/kernel/thread.c#L281

Reply:
that function just calls into this function